contrib/src/main/java/com/datatorrent/contrib/kinesis/AbstractKinesisInputOperator.java

```diff
@@ -507,9 +514,8 @@ public void endWindow()
   @Override
   public void activate(OperatorContext ctx)
   {
-    if(isReplayState)
-    {
-      // If it is a replay state, don't start the consumer
+    // If it is a replay state, don't start the consumer
+    if (context.getValue(OperatorContext.ACTIVATION_WINDOW_ID) < windowDataManager.getLargestCompletedWindow()) {
```

Comment:
Verify the below condition the also:
context.getValue(OperatorContext.ACTIVATION_WINDOW_ID) != Stateless.WINDOW_ID